packages/fast-components-react-base/src/stack-panel/stack-panel.tsx

```diff
@@ -0,0 +1,738 @@
+        }
+
+        this.itemPositions = itemPositions;
+        this.itemContainerSpan =
+            this.itemPositions.length > 0 ? this.itemPositions[itemCount - 1].end : 0;
+    };
+
+    /**
+     *  Get the size of the viewport along the panel's axis
+     */
+    private getViewportSpan = (): number => {
+        if (isNil(this.rootElement.current)) {
+            return 0;
+        }
+        if (this.props.orientation === stackPanelOrientation.horizontal) {
+            return this.rootElement.current.clientWidth;
+        } else {
+            return this.rootElement.current.clientHeight;
+        }
+    };
+
+    /**
+     * Request's an animation frame if there are currently no open animation frame requests
+     */
+    private requestFrame = (): void => {
+        if (this.openRequestAnimationFrame === null) {
+            this.openRequestAnimationFrame = window.requestAnimationFrame(
+                this.gotAnimationFrame
+            );
+        }
+    };
+
+    /**
+     * Got an animation frame
+     */
+    private gotAnimationFrame = (): void => {
+        this.openRequestAnimationFrame = null;
+        this.updateLayout();
+        if (this.isScrollAnimating) {
+            this.updateScrollAnimation();
+        }
+    };
+
+    /**
+     * Recalculates the layout
+     */
+    private updateLayout = (): void => {
+        if (isNil(this.rootElement.current) || isNil(this.itemContainerElement.current)) {
+            return;
+        }
+
+        if (this.props.orientation === stackPanelOrientation.horizontal) {
+            this.updateDirection();
+        }
+
+        const scrollPos: number = this.getScrollPosition();
+        const lastIndex: number =
+            this.itemPositions.length === 0 ? 0 : this.itemPositions.length - 1;
+
+        const effectiveViewportSpan: number =
+            this.viewPortSpan >= this.props.defaultItemSpan
+                ? this.viewPortSpan
+                : this.props.defaultItemSpan;
+
+        const visibleRangeStartIndex: number = this.getThresholdItemIndex(
+            this.itemPositions,
+            0,
+            scrollPos
+        );
+        const visibleRangeEndIndex: number = this.getThresholdItemIndex(
+            this.itemPositions,
+            visibleRangeStartIndex,
+            scrollPos + effectiveViewportSpan
+        );
+
+        const bufferLength: number = this.getBufferLength();
+
+        let renderStartIndex: number = visibleRangeStartIndex - bufferLength;
+        if (renderStartIndex < 0) {
+            renderStartIndex = 0;
+        }
+
+        let renderEndIndex: number = visibleRangeEndIndex + bufferLength;
+        if (renderEndIndex > lastIndex) {
+            renderEndIndex = lastIndex;
+        }
+        this.setState({
+            renderedRangeStartIndex: renderStartIndex,
+            renderedRangeEndIndex: renderEndIndex,
+            isScrolling: this.itemContainerSpan > this.viewPortSpan ? true : false,
```

Comment:
can just be `this.itemContainerSpan > this.viewPortSpan`, don't need a ternary for true/false.